packages/client/tests/functional/relation-load-strategy/supported-queries.ts

```diff
@@ -0,0 +1,561 @@
+          },
+          select: {
+            login: true,
+            posts: {
+              select: {
+                title: true,
+                comments: {
+                  select: {
+                    author: {
+                      select: {
+                        login: true,
+                      },
+                    },
+                    body: true,
+                  },
+                },
+              },
+            },
+          },
+        }),
+      ).resolves.toMatchInlineSnapshot(`
+        {
+          login: author,
+          posts: [
+            {
+              comments: [
+                {
+                  author: {
+                    login: commenter,
+                  },
+                  body: a comment,
+                },
+              ],
+              title: first post,
+            },
+          ],
+        }
+      `)
+
+      await expectQueryCountAtLeast({
+        join: 1,
+        query: 4,
+      })
+
+      expectLateralJoinToBeUsedIfRequested()
+    })
+
+    test('findUniqueOrThrow', async () => {
+      await expect(
+        prisma.user.findUniqueOrThrow({
+          relationLoadStrategy,
+          where: {
+            login: 'author',
+          },
+          select: {
+            login: true,
+            posts: {
+              select: {
+                title: true,
+                comments: {
+                  select: {
+                    author: {
+                      select: {
+                        login: true,
+                      },
+                    },
+                    body: true,
+                  },
+                },
+              },
+            },
+          },
+        }),
+      ).resolves.toMatchInlineSnapshot(`
+        {
+          login: author,
+          posts: [
+            {
+              comments: [
+                {
+                  author: {
+                    login: commenter,
+                  },
+                  body: a comment,
+                },
+              ],
+              title: first post,
+            },
+          ],
+        }
+      `)
+
+      await expectQueryCountAtLeast({
+        join: 1,
+        query: 4,
+      })
+
+      expectLateralJoinToBeUsedIfRequested()
+    })
+
+    test('create', async () => {
+      await expect(
+        prisma.user.create({
+          relationLoadStrategy,
+          data: {
+            login: 'reader',
+            comments: {
+              create: {
+                post: {
+                  connect: { id: postId },
+                },
+                body: 'most insightful indeed!',
+              },
+            },
+          },
+          select: {
+            login: true,
+            comments: {
+              select: {
+                post: {
+                  select: {
+                    title: true,
+                  },
+                },
+                body: true,
+              },
+            },
+          },
+        }),
```

Comment:
I can add one while we're waiting for the engines to be released anyway

Reply:
I took the liberty of interpreting "without any `select`" as "`select` with a single scalar". Create without any selection does not exist in Prisma, if you don't specify `select` Prisma Client will implicitly select all scalars, and that's problematic for snapshots in these tests because IDs aren't stable here.